client.go

```diff
@@ -328,8 +375,50 @@ func (a *Client) CopyFromRemotePassThru(ctx context.Context, w io.Writer, remote
 			errCh <- errors.New(res.GetMessage())
 			return
 		}
+		if res.NoStandardProtocolType() {
+			errCh <- errors.New(fmt.Sprintf("Input from server doesn't follow protocol: %s", res.GetMessage()))
+			return
+		}
+
+		fileInfo := NewFileInfos()
+
+		if res.IsTime() {
+			timeInfo, err := res.ParseFileTime()
+
+			if err != nil {
+				errCh <- err
+				return
+			}
+
+			fileInfo.Update(timeInfo)
+
+			res, err = ParseResponse(r)
```

Comment:
I think we would need to send `Ack` first before reading the response, otherwise the remote might not proceed with the next message.

Reply:
I haven't implemented the -p functionality yet. I did it like this because on the custom ssh server(implemented by wish) the T and C message come one after another without waiting for an Ack. When you use -p, then you're right, the Ack has to be there. I thought of another fix where I read and process until the buffer is ready.